notification/pushbullet.py

```diff
@@ -0,0 +1,174 @@
+        required: false
+        default: null
+    push_type:
+        description:
+          - Thing you wish to push.
+        required: false
+        default: note
+        choices: [ "note", "link" ]
+    title:
+        description:
+          - Title of the notification.
+        required: true
+    body:
+        description:
+          - Body of the notification, e.g. Details of the fault you're alerting.
+        required: false
+
+notes:
+   - Requires pushbullet.py Python package on the remote host.
+     You can install it via pip with ($ pip install pushbullet.py).
+     See U(https://github.com/randomchars/pushbullet.py)
+'''
+
+EXAMPLES = '''
+# Sends a push notification to a device
+- pushbullet: 
+    api_key: "ABC123abc123ABC123abc123ABC123ab"
+    device: "Chrome"
+    title: "You may see this on Google Chrome"
+
+# Sends a link to a device
+- pushbullet: 
+    api_key: "ABC123abc123ABC123abc123ABC123ab"
+    device: "Chrome"
+    push_type: "link"
+    title: "Ansible Documentation"
+    body: "http://docs.ansible.com/"
+
+# Sends a push notification to a channel
+- pushbullet: 
+    api_key: "ABC123abc123ABC123abc123ABC123ab"
+    channel: "my-awesome-channel"
+    title: "Broadcasting a message to the #my-awesome-channel folks"
+
+# Sends a push notification with title and body to a channel
+- pushbullet: 
+    api_key: "ABC123abc123ABC123abc123ABC123ab"
+    channel: "my-awesome-channel"
+    title: "ALERT! Signup service is down"
+    body: "Error rate on signup service is over 90% for more than 2 minutes"
+'''
+
+try:
+    from pushbullet import PushBullet
+except ImportError:
+    pushbullet_found = False
+else:
+    pushbullet_found = True
+
+# ===========================================
+# Main
+#
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            api_key     = dict(type='str', required=True),
+            channel     = dict(type='str', default=None),
+            device      = dict(type='str', default=None),
+            push_type   = dict(type='str', default="note", choices=['note', 'link']),
+            title       = dict(type='str', required=True),
+            body        = dict(type='str', default=None)
+        ),
+        supports_check_mode=True
+    )
+
+    api_key     = module.params['api_key']
+    channel     = module.params['channel']
+    device      = module.params['device']
+    push_type   = module.params['push_type']
+    title       = module.params['title']
+    body        = module.params['body']
+
+    if not pushbullet_found:
+        module.fail_json(msg="Python 'pushbullet.py' module is required. Install via: $ pip install pushbullet.py")
+
+    # Init pushbullet 
+    pb = PushBullet(api_key)
+    target = None
+
+    # Checks for channel/device
+    if device is not None and channel is not None:
+        module.fail_json(msg="You can't use both device and channel at the same time.")
+
+    if device is None and channel is None:
```

Comment:
this is ok. just wanted to point out there is a option for AnsibleModule:

~~~
mutually_exclusive = (
    ['device', 'channel'],
),
~~~